File: src/main/java/org/lanternpowered/server/advancement/criteria/OrCriterionFactory.java
/*
 * Lantern
 *
 * Copyright (c) LanternPowered <https://www.lanternpowered.org>
 * Copyright (c) SpongePowered <https://www.spongepowered.org>
 * Copyright (c) contributors
 *
 * This work is licensed under the terms of the MIT License (MIT). For
 * a copy, see 'LICENSE.txt' or <https://opensource.org/licenses/MIT>.
 */
package org.lanternpowered.server.advancement.criteria;

import org.spongepowered.api.advancement.criteria.AdvancementCriterion;
import org.spongepowered.api.advancement.criteria.OrCriterion;

import java.util.Arrays;

public final class OrCriterionFactory implements OrCriterion.Factory {

    @Override
    public AdvancementCriterion of(AdvancementCriterion... criteria) {
        return AbstractCriterion.build(OrCriterion.class, LanternOrCriterion::new, Arrays.asList(criteria));
    }

    @Override
    public AdvancementCriterion of(Iterable<AdvancementCriterion> criteria) {
        return AbstractCriterion.build(OrCriterion.class, LanternOrCriterion::new, criteria);
    }
}

File: src/main/kotlin/org/lanternpowered/server/advancement/criteria/OrCriterionFactory.kt
/*
 * Lantern
 *
 * Copyright (c) LanternPowered <https://www.lanternpowered.org>
 * Copyright (c) SpongePowered <https://www.spongepowered.org>
 * Copyright (c) contributors
 *
 * This work is licensed under the terms of the MIT License (MIT). For
 * a copy, see 'LICENSE.txt' or <https://opensource.org/licenses/MIT>.
 */
package org.lanternpowered.server.advancement.criteria

import org.spongepowered.api.advancement.criteria.AdvancementCriterion
import org.spongepowered.api.advancement.criteria.OrCriterion

object OrCriterionFactory : OrCriterion.Factory {

    override fun of(vararg criteria: AdvancementCriterion): AdvancementCriterion =
            AbstractCriterion.build(OrCriterion::class.java, criteria.asSequence(), ::LanternOrCriterion)

    override fun of(criteria: Iterable<AdvancementCriterion>): AdvancementCriterion =
            AbstractCriterion.build(OrCriterion::class.java, criteria.asSequence(), ::LanternOrCriterion)
}
